app/javascript/components/workflow-credential-mapping-form/workflow-credential-mapping-form.schema.js

```diff
@@ -0,0 +1,162 @@
+        options: Object.keys(payloadCredentials).map((key) => ({
+          value: key,
+          label: key,
+        })),
+        resolveProps: (props, { meta, input }, formOptions) => {
+          // This ensures credentialReferences is set back to undefined when credential_references is reset to default
+          if (meta.pristine && credentialReferences !== undefined) {
+            setState((state) => ({
+              ...state,
+              credentialReferences: undefined,
+            }));
+          }
+
+          /*
+            Anytime the user selects a credential reference to map check to see if it's already mapped.
+            If it is, then update 'workflow_credentials' and 'credential_field' to display this mapping.
+            Otherwise reset them to empty.
+          */
+          if (credentials[input.value] && credentialReferences !== input.value) {
+            formOptions.change('workflow_credentials', credentials[input.value].credential_ref);
+            formOptions.change('credential_field', credentials[input.value].credential_field);
+            setState((state) => ({
+              ...state,
+              workflowCredentials: credentials[input.value].credential_ref,
+              credentialField: credentials[input.value].credential_field,
+            }));
+          } else if (credentialReferences !== (input.value || undefined)) {
+            formOptions.change('workflow_credentials', '');
+            formOptions.change('credential_field', '');
+            setState((state) => ({
+              ...state,
+              workflowCredentials: '',
+              credentialField: '',
+            }));
+          }
+        },
+        onChange: (value) => setState((state) => ({
+          ...state,
+          credentialReferences: value,
+        })),
+      },
+      {
+        component: componentTypes.SUB_FORM,
+        name: 'workflow_credentials_section',
+        id: 'workflow_credentials_section',
+        condition: { when: 'credential_references', isNotEmpty: true },
+        fields: [
+          {
+            component: componentTypes.SELECT,
+            name: 'workflow_credentials',
+            label: __('Workflow Credential'),
+            placeholder: __('<Choose>'),
+            includeEmpty: true,
+            options: workflowAuthentications,
+            onChange: (value) => {
+              setState((state) => ({ ...state, workflowCredentials: value }));
+              mapCredentials();
+            },
+          },
+          {
+            component: componentTypes.SELECT,
+            name: 'credential_field',
+            label: __('Credential Field'),
+            placeholder: __('<Choose>'),
+            includeEmpty: true,
+            options: [
+              { label: 'userid', value: 'userid' },
+              { label: 'password', value: 'password' },
+              { label: 'auth_key', value: 'auth_key' },
+              { label: 'auth_key_password', value: 'auth_key_password' },
```

Comment:
The label here should match whatever is in the credential creation page so that it's consistent.  It should also be translated.

```suggestion
              { label: 'userid', value: 'userid' },
              { label: 'password', value: 'password' },
              { label: 'auth_key', value: 'auth_key' },
              { label: 'auth_key_password', value: 'auth_key_password' },
```

Reply:
Also, we should _only_ allow those fields on submit.  That is, we should fail if someone tries to hack the form and send anything but these fields.

Reply:
Not sure I'm following here